### PR TITLE
Add new operators for table operations, fix outputs of openai eval op…

### PR DESCRIPTION
**Pull request by sourabhagr**
…erators, check if table already exists for writers, make eval name configurable for model grade operator

### Type of Change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Description

Please include a summary of the change and which issue it addresses. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

### Checklist

- [ ] I have read the [CONTRIBUTING](https://github.com/uptrain-ai/uptrain/blob/main/CONTRIBUTING.md) document.
- [ ] My code follows the code style (BLACK) of this project.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have updated the documentation accordingly.
- [ ] I have added an appropriate CHANGELOG entry.

### Author's Note

Please provide any additional information, questions, or concerns you have about this pull request.

### CODE REVIEW

File: uptrain/operators/language/openai_evals.py [hunk 1]
```diff
@@ -127,18 +129,17 @@ def run(self, data: t.Optional[pl.DataFrame] = None) -> TYPE_OP_OUTPUT:
...
+        }
+        extra.update(type_data)
         return {
-            "output": output_data,  # events are of multiple kinds. Undecided how to handle them in a consistent schema yet
-            "extra": {
-                "all_events": recorder.get_list_events(),
-                "run_data": recorder.get_run_data(),
-                "final_report": to_py_types(final_report),
-            },
+            "output": None,  # events are of multiple kinds. Undecided how to handle them in a consistent schema yet
```
Comment (ananis25) [inline uptrain/operators/language/openai_evals.py:141]: How do we store the output of this operator if it is None? SimpleCheck doesn't check for data in the "extra" key.

**Review by ananis25 — CHANGES_REQUESTED** [review]

File: uptrain/io/writers.py [hunk 2]
```diff
@@ -35,10 +40,12 @@ def __init__(self, op: DeltaWriter):
         self.columns = list(self.op.columns) if self.op.columns is not None else None
 
     def run(self, data: pl.DataFrame) -> TYPE_OP_OUTPUT:
+        if os.path.exists(self.op.fpath):
```
[the remark this answers is not in the document]
Comment (sourabhagr) [reply uptrain/io/writers.py:43]: Makes sense, will remove this

File: uptrain/io/writers.py [hunk 3]
```diff
@@ -60,6 +69,8 @@ def __init__(self, op: JsonWriter):
...
```
[the remark this answers is not in the document]
Comment (sourabhagr) [reply uptrain/io/writers.py:72]: Makes sense, raising an alert

File: uptrain/operators/language/openai_evals.py (hunk 1 above)
Comment (sourabhagr) [reply uptrain/operators/language/openai_evals.py:141]: So, the output of this operator is a bit complicated as it has two kinds of information - 
1) Prompt-Response
2) Matching/Metrics calculation
For now, updated output to reflect both the information. Ideally, this operator should not be used in SimpleCheck?

File: uptrain/io/writers.py [hunk 4]
```diff
@@ -1,12 +1,15 @@
...
+import os
+import shutil
 
 from pydantic import BaseModel
 import polars as pl
 import deltalake as dl
 from pathlib import Path
 
 from uptrain.operators.base import *
+from uptrain.io.readers import *
```
[the remark this answers is not in the document]
Comment (sourabhagr) [reply uptrain/io/writers.py]: Makes sense, done

File: uptrain/operators/language/openai_evals.py (hunk 1 above)
Comment (ananis25) [reply uptrain/operators/language/openai_evals.py]: Makes sense. It can be called an `OpenaiCheck` or something.

**Review by ananis25 — APPROVED** [review]

**Pull request merged**